New structure

rjuliaros opened the pull request:
It is proposed to change the launch mode of the simulation, to make it closer to reality. The simulation is splitted into three parts:
+ Gazebo Environment (run_simulation.launch contains everything related to the simulation environment)
+ Rviz Config (run_rviz.launch contains a default configuration for RViz)
+ Run Robot (run_robot.launch to launch the robot, with or without the arm)
+ These files are launched by rbsherpa_complete.launch.

rjuliaros committed:
Created files to separate complete.launch into run_simulation, run_rviz and run_robot

rjuliaros committed:
Created files to separate complete.launch into run_simulation, run_rviz and run_robot

rjuliaros committed:
Corrected file's name of the .rviz

rjuliaros committed:
Changed id_robot -> robot_id

rjuliaros committed:
Fixed errors: (...)
* id_robot->robot_id
*Removed harcoded frames
*Removed unused args

rjuliaros committed:
Corrected errors with rviz node and fixed minor errors

rjuliaros committed:
Merge branch 'new-structure' of https://github.com/RobotnikAutomation/rbsherpa_sim into new-structure

# Conflicts:
#	rbsherpa_sim_bringup/launch/run_robot.launch
#	rbsherpa_sim_bringup/launch/run_rviz.launch

rjuliaros committed:
Add .yaml with controllers

rjuliaros committed:
Delete unused .launch

rjuliaros committed:
two_arm_control.yaml : Change ns of each arm group to be closest as possible of real-robot

Namespace for each arm nested into arm/(left or right)arm/pos_traj_controller/...

rjuliaros committed:
run_robot.launch: Add params to spawn and control the arm

rjuliaros committed:
updated readme.md to add bi-arm explanaiton

rjuliaros committed:
rbsherpa_complete.launch: now it runs sim using run_robot/sim/rviz

rjuliaros committed:
run_robot.launch: solved naming issueand erased rviz call.

rjuliaros committed:
run_rviz.launch: Solved naming issue

rjuliaros committed:
run_simulation.launch: solved naming issue

rjuliaros committed:
arm_control.yaml: Moved to rbsherpa_common package

rjuliaros committed:
rbsherpa_completa&run_robot.launch: removed inconsistent parameter.

rjuliaros committed:
reamde.md: Update launch options

David-rn commented:
First of all, great job updating all the files to the new structure. Each group of changes are related to one commit, which is good, and they have descriptive senteces. However, there are some issues (not a big deal) to correct: 

In the Readme file there are a couple of adjustments to make:
- `roslaunch rbsherpa_sim_bringup rbsherpa_complete.launch kinematics:=omni twist2ackermann:=false` -> Does not work, I think the parameter `arm_controllers` is missing
- `roslaunch rbsherpa_sim_bringup rbsherpa_complete.launch robot_xacro:=rbsherpa_xl_std.urdf.xacro` -> The URDF does not exist
- `roslaunch rbsherpa_xl_2ur10_moveit_config rbsherpa_xl_moveit_config.launch` -> It is missing the namespace ROS_NAMESPACE=robot
- `roslaunch rbsherpa_moveit_ur10 demo.launch` -> The same as before, the namespace is missing
- When setting the parameter `launch_move:=True` the simulation does not start because the launch file is missing


Lastly, it is important to have in mind that `launch_move_base` will launch the move_base package but it needs a map and a localization to work with, so if you are planing to work with that package, then you will need to add them to the simulation.

Great work! :smile:

rjuliaros committed:
run_robot.launch: removed move_base and move launch files

+ The intention is that the simulation be as similar as possible to the real robot.
  So the navigation and perception nodes should not be launched in run_robot.launch
+ Issue has been created to solve this problem.

rjuliaros committed:
rbsherpa_complete.launch: removed launch_move and launch_move_base params

+ These params are unused

rjuliaros committed:
run_robot.launch: removed launch_move and launch_move_base params

+ These params are unused.

rjuliaros committed:
README.md: updated readme

rjuliaros committed:
README.md: Updated readme

David-rn commented:
The changes look great, I'm going to merge the new structure with master. 

Good job! :+1:
